### PR TITLE
Fix indexing problems with findc and saveme

### PR DESCRIPTION
**Pull request by JonLamy**
Also added error message handling for those commands when task index is out of bounds.

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/FindContactCommandParserTest.java [hunk 1]
```diff
@@ -26,6 +26,17 @@ public void parse_invalidArgs_throwsParseException() {
         // multiple search prefixes (e.g. both n/ and m/)
         assertParseFailure(parser, " n/Alice m/CS1101S",
                 String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindContactCommand.MESSAGE_USAGE));
+
+        // invalid prefixes
+        assertParseFailure(parser, "s/ foo",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindContactCommand.MESSAGE_USAGE));
+
+        assertParseFailure(parser, "s/ foo g/ bar",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindContactCommand.MESSAGE_USAGE));
```
Comment (parnikkapore) [inline src/test/java/seedu/address/logic/parser/FindContactCommandParserTest.java:35]: In general, invalid prefixes are specified to be ignored in the UG.

Also, is there a reason this is still using the space-after-prefix syntax?

**Review by parnikkapore — COMMENTED** [review]
Looking good! ~~Most~~ All of the suggestions here are stylistic.

File: src/main/java/seedu/address/logic/commands/FindContactCommand.java [hunk 2]
```diff
@@ -40,8 +43,17 @@ public static FindContactCommand withoutArgs() {
...
+    public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
+
+        if (predicate instanceof CanHelpWithTaskPredicate) {
+            CanHelpWithTaskPredicate taskPredicate = (CanHelpWithTaskPredicate) predicate;
+            List<Task> lastShownList = model.getSortedTaskList();
+            if (taskPredicate.getTaskIndex().getZeroBased() >= lastShownList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+            }
+        }
```
[the remark this answers is not in the document]
Comment (JonLamy) [reply src/main/java/seedu/address/logic/commands/FindContactCommand.java:55]: I thought so too, but it seems that too large a index isn't necessarily a invalid input, but more of a execution error! (also it follows logic in commands like EditContactComman)

File: src/main/java/seedu/address/model/person/CanHelpWithTaskPredicate.java [hunk 3]
```diff
@@ -56,7 +57,7 @@ public boolean equals(Object other) {
         System.out.println(other);
         return other == this // short circuit if same instance
                 || (other instanceof CanHelpWithTaskPredicate
-                && taskIndex == ((CanHelpWithTaskPredicate) other).taskIndex);
+                && taskIndex.getZeroBased() == ((CanHelpWithTaskPredicate) other).taskIndex.getZeroBased());
```
[the remark this answers is not in the document]
Comment (JonLamy) [reply src/main/java/seedu/address/model/person/CanHelpWithTaskPredicate.java:60]: Nice catch, fixed!

File: src/test/java/seedu/address/logic/parser/FindContactCommandParserTest.java (hunk 1 above)
Comment (JonLamy) [reply src/test/java/seedu/address/logic/parser/FindContactCommandParserTest.java:35]: No reason, but I've changed to no space for consistency

File: src/test/java/seedu/address/logic/commands/FindContactCommandTest.java [hunk 4]
```diff
@@ -76,6 +79,28 @@ public void execute_multipleKeywords_multiplePersonsFound() {
...
+
+    @Test
+    public void execute_searchByInvalidTask_throwsCommandException() {
+        CanHelpWithTaskPredicate predicate = new CanHelpWithTaskPredicate(100);
+        FindContactCommand command = new FindContactCommand(predicate);
+        assertCommandFailure(command, model, Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void withoutArgs_noArgs_returnFindContactByTaskCommand() {
```
[the remark this answers is not in the document]
Comment (JonLamy) [reply src/test/java/seedu/address/logic/commands/FindContactCommandTest.java:99]: Agreed

File: src/main/java/seedu/address/logic/commands/FindContactCommand.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (parnikkapore) [reply src/main/java/seedu/address/logic/commands/FindContactCommand.java:55]: Ah, I see.

**Review by danzzzerl — APPROVED** [review]
LGTM

**Pull request merged**